unity-environment/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -583,6 +591,23 @@ public virtual void CollectObservations()
 
         }
 
+        protected void MaskDiscreteAction(bool[] mask)
+        {
+            if (brain.brainParameters.vectorActionSpaceType == SpaceType.continuous)
+            {
+                throw new UnityAgentsException(
+                    "Invalid : Cannot set a mask for actions with Continuous Control.");
```

Comment:
This error should maybe be something like "Cannot use action mask with continuous action space."

Reply:
Maybe the mask can also apply to continuous action space? For example we could limit the action to be only positive (a mask of (0 - positive infinity)), so that in 3dball it will only turn in a certain direction given certain state.